core/blockchain.go

```diff
@@ -2375,40 +2382,20 @@ func (bc *BlockChain) UpdateValidatorVotingPower(
 			}
 		}
 
-		// This means it's already in staking epoch
+		// This means it's already in staking epoch, and
+		// compute APR for validators in current committee only
 		if currentEpochSuperCommittee.Epoch != nil {
-			wrapper, err := state.ValidatorWrapper(key)
-			if err != nil {
-				return nil, err
-			}
+			if _, ok := existing.LookupSet[key]; ok {
+				wrapper, err := state.ValidatorWrapper(key)
+				if err != nil {
+					return nil, err
+				}
 
-			if wrapper.Delegations[0].Amount.Cmp(common.Big0) > 0 {
```

Comment:
no longer need this check, as we are using snapshot total delegation and there is a check inside `ComputeForValidator` to ensure no div-by-zero happens.